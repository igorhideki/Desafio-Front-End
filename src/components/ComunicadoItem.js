import React from 'react';
import PropTypes from 'prop-types';

const ComunicadoItem = (props) => (
  <li>
    <h4>{props.title}</h4>
    <span>{props.type}</span>
    <span>{props.date}</span>
    <p>{props.description}</p>
  </li>
);

ComunicadoItem.propTypes = {
  id: PropTypes.number.isRequired,
  title: PropTypes.string.isRequired,
  seen: PropTypes.bool.isRequired,
  type: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  fullDescription: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  file: PropTypes.shape({
    type: PropTypes.string,
    name: PropTypes.string,
    url: PropTypes.string,
  })
};

export default ComunicadoItem;

