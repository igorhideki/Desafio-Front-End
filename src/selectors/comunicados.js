import moment from 'moment';

const getVisiblesComunicados = (comunicados, { type = -1 }) => {
  const formatDate = 'YYYY-MM-DD HH:mm';
  return comunicados.filter((comunicado) => {
    if (type === -1) {
      return comunicado;
    }
    return comunicado.type === type;
  }).sort((a, b) => {
    return moment(a.date, formatDate).format() < moment(b.date, formatDate).format() ? 1 : -1;
  });
};

export default getVisiblesComunicados;

