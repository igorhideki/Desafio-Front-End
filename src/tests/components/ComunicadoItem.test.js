import React from 'react';
import { shallow } from 'enzyme';
import ComunicadoItem from '../../components/ComunicadoItem';
import comunicados from '../fixtures/comunicados';

let wrapper, history;

beforeAll(() => {
  history = { push: jest.fn() };
  wrapper = shallow(<ComunicadoItem.WrappedComponent key={comunicados[1].id} {...comunicados[1]} history={history}/>);
});

test('should render ComunicadoItem correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle clicked in comunicadoItem', () => {
  const value = `details/${comunicados[1].id}`;
  wrapper.find('li').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith(value);
});
