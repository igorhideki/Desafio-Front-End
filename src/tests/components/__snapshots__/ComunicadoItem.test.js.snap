// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ComunicadoItem correctly 1`] = `
<li
  className="comunicados__item  comunicados__item--blue"
  onClick={[Function]}
>
  <div
    className="comunicados__row"
  >
    <span
      className="comunicados__title"
    >
      Bônus Fevereiro
    </span>
    <span
      className="comunicados__id"
    >
      2
    </span>
  </div>
  <div
    className="comunicados__row"
  >
    <span
      className="comunicados__type"
    >
      Força de Vendas
    </span>
    <span
      className="comunicados__date"
    >
      02/05/2018 - 10h30
    </span>
  </div>
</li>
`;
